src/transactions/ManageOfferOpFrame.cpp

```diff
@@ -472,4 +472,36 @@ ManageOfferOpFrame::buildOffer(AccountID const& account,
     o.flags = flags;
     return o;
 }
+
+std::unordered_set<LedgerKey>
+ManageOfferOpFrame::getLedgerKeysToPrefetch(Application& app)
+{
+    // TODO (mlo) will need to deal with best offers as well
+
+    std::unordered_set<LedgerKey> keys;
+    if (mManageOffer.amount == 0)
+    {
+        return keys;
+    }
+
+    // Prefetch existing offer
+    if (mManageOffer.offerID)
+    {
+        keys.insert(getLedgerKey(getSourceID(), mManageOffer.offerID));
```

Comment:
Also better to use `emplace` for all of these (not going to write this comment again, but it applies to other analogous code as well).